### PR TITLE
Fix error when clicking on table header

### PR DESCRIPTION
**Pull request by obniavko**
dev
## ZeroWaste project

* [Project ticket #705](https://github.com/ita-social-projects/ZeroWaste/issues/705)
* [Project ticket #731](https://github.com/ita-social-projects/ZeroWaste/issues/731)

## Code reviewers

- [x] @kisiohlova 
- [ ] @loqimean 

## Summary of issue

An error occurs when clicking on table header on [Users](https://zero-waste-staging.onrender.com/en/account/users)/[Products](https://zero-waste-staging.onrender.com/en/account/products)/[Categories](https://zero-waste-staging.onrender.com/en/account/categories) pages.

## Summary of change

1. Fixed error.
sorted
3. Added ability to sort items for [Messages](https://zero-waste-staging.onrender.com/en/account/messages) page.
4. Added ability to sort items for [Calculators](https://zero-waste-staging.onrender.com/en/calculators) page.
calculators

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -22,4 +22,8 @@ class Category < ApplicationRecord
   scope :ordered_by_name, -> { order(:name) }
   scope :ordered_by_priority, -> { order(:priority) }
   scope :unsigned_categories, ->(product) { where.not(id: product.categories_by_prices) }
+
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (loqimean) [inline app/models/category.rb:26]: бачу тут не каверид строчки)

File: app/models/message.rb [hunk 2]
```diff
@@ -12,7 +12,13 @@
 #  updated_at :datetime         not null
 #
 class Message < ApplicationRecord
+  scope :ordered_by_title, -> { order(:title) }
+
   validates :title, presence: true, length: { minimum: 5 }
   validates :message, presence: true, length: { minimum: 20 }
   validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
+
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (loqimean) [inline app/models/message.rb:21]: те ж саме

File: app/models/product.rb [hunk 3]
```diff
@@ -47,6 +47,10 @@ def build_unsigned_categories
     prices.build(unsigned_categories.map { |category| { category: category } })
   end
 
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (loqimean) [inline app/models/product.rb:50]: каверидж просів

File: app/models/user.rb [hunk 4]
```diff
@@ -132,4 +132,8 @@ def self.new_with_session(params, session)
   def full_name
     [first_name, last_name].compact_blank.join(" ")
   end
+
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (loqimean) [inline app/models/user.rb:136]: теж просів

File: spec/requests/account/messages_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+      expect(response).to be_successful
+      expect(response).to render_template(:index)
+      expect(response.body).to include(message.title)
+    end
+  end
+
+  describe "GET :show" do
+    it "is successful" do
+      get account_message_path(message)
```
Comment (loqimean) [inline spec/requests/account/messages_spec.rb:20]: напиши пліз тести на всі сторінки, які мали ерори

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/models/category.rb (hunk 1 above)
Comment (obniavko) [reply app/models/category.rb:26]: Fixed

File: app/models/message.rb (hunk 2 above)
Comment (obniavko) [reply app/models/message.rb:21]: Fixed

File: app/models/product.rb (hunk 3 above)
Comment (obniavko) [reply app/models/product.rb:50]: Fixed

File: app/models/user.rb (hunk 4 above)
Comment (obniavko) [reply app/models/user.rb:136]: Fixed

File: spec/requests/account/messages_spec.rb (hunk 5 above)
Comment (obniavko) [reply spec/requests/account/messages_spec.rb:26]: Done

File: spec/requests/account/users_spec.rb [hunk 6]
```diff
@@ -27,6 +27,11 @@
       expect(csv_content).to match(user.last_name)
       expect(csv_content).to match(user.last_sign_in_at.to_s)
     end
+
+    it "returns the expected attributes" do
+      get account_users_path
```
Comment (loqimean) [inline spec/requests/account/users_spec.rb:32]:
```suggestion
      get account_users_path

```

File: spec/requests/account/users_spec.rb [hunk 7]
```diff
@@ -27,6 +27,11 @@
...
+      expect(User.ransackable_attributes).to eq(["created_at", "id", "blocked", "country", "email", "first_name", "last_name", "updated_at"])
```
Comment (loqimean) [inline spec/requests/account/users_spec.rb:33]: що це за тест взагалі такий?)

File: spec/requests/categories_spec.rb [hunk 8]
```diff
@@ -14,6 +14,12 @@
 
       expect(response).to be_successful
     end
+
+    it "returns the expected attributes" do
+      get account_categories_path
+
+      expect(Category.ransackable_attributes).to eq(["created_at", "id", "name", "priority", "updated_at"])
```
Comment (loqimean) [inline spec/requests/categories_spec.rb:21]: давай міняй

Comment (loqimean) [reply spec/requests/categories_spec.rb:21]: я казав ітератором пройтися і зробити запити з цими параметрами)

**Review by loqimean — CHANGES_REQUESTED** [review]

File: spec/requests/categories_spec.rb (hunk 8 above)
Comment (obniavko) [reply spec/requests/categories_spec.rb]: Fixed

**Review by loqimean — APPROVED** [review]

**Pull request merged**